hack/tools/check-go-version.sh

```diff
@@ -0,0 +1,80 @@
+#!/bin/bash
+
```

Comment:
I know this is just a copy, but can you add a comment in here that the original source of this is o-f/op-con and to keep them in sync if there is a change?
We have the same kind of standard [here](https://github.com/operator-framework/operator-registry/blob/master/.golangci.yaml#L1-L14).